### ISSUE TITLE
`require-context-role` false positive for table role

### ISSUE THREAD

**Issue by drewlee** [ember-template-lint/ember-template-lint#2442]
The `table` role is a valid context role for the `row` and `rowgroup` roles as outlined in the [WAI-ARIA 1.1 specification](https://www.w3.org/TR/wai-aria-1.1/#roles). However, it gets flagged as an issue using ember-template-lint v4.3.0.

For example, the following markup:
```html
<div role="table">
  <div role="row">
    <div role="cell">One</div>
    <div role="cell">Two</div>
  </div>
</div>
```
Results in the following error:
```sh
app/templates/application.hbs
  8:7  error  You have an element with the role of "row" but it is missing the required (immediate) parent element of "[grid, rowgroup, treegrid]". Reference: https://www.w3.org/TR/wai-aria-1.0/roles#row.  require-context-role
```
[Table](https://www.w3.org/TR/wai-aria-1.1/#table) should be an acceptable role in addition to `grid`, `rowgroup`, and `treegrid`.

In addition to fixing this bug, we should check the other roles against WAI-ARIA 1.1 to ensure there are no other false positives.
